### PR TITLE
add architecture and previews guide to docs menu sections

### PR DESCRIPTION
**Pull request by erquhart**
As mentioned in https://github.com/netlify/netlify-cms/pull/1366#issuecomment-390075964.

### CODE REVIEW

**Comment by verythorough** [comment]
Deploy preview for *cms-demo* ready!

Built with commit 2e049792c9b4f45b03238d792d4b60d027353732

https://deploy-preview-1368--cms-demo.netlify.com

**Comment by verythorough** [comment]
Deploy preview for *netlify-cms-www* ready!

Built with commit 2e049792c9b4f45b03238d792d4b60d027353732

https://deploy-preview-1368--netlify-cms-www.netlify.com

File: website/site/content/docs/customization.md [hunk 1]
```diff
@@ -1,6 +1,9 @@
 ---
 title: Custom Previews
```
Comment (tech4him1) [inline website/site/content/docs/customization.md]: We should change the form of the titles to be the same: "Creating Custom Previews" would match "Creating Custom Widgets".

Comment (erquhart) [reply website/site/content/docs/customization.md]: Good point.

**Comment by verythorough** [comment]
LGTM!

**Pull request merged**